source/common/grpc/async_client_impl.cc

```diff
@@ -141,11 +141,16 @@ void AsyncStreamImpl::onHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_s
 
 void AsyncStreamImpl::onData(Buffer::Instance& data, bool end_stream) {
   decoded_frames_.clear();
-  if (!decoder_.decode(data, decoded_frames_)) {
+  if (decoder_.decode(data, decoded_frames_).code() == absl::StatusCode::kInternal) {
     streamError(Status::WellKnownGrpcStatus::Internal);
     return;
   }
 
+  if (decoder_.decode(data, decoded_frames_).code() == absl::StatusCode::kResourceExhausted) {
```

Comment:
Is this right that the `decode` method is now called twice?

Reply:
Nice catch! Yes, I should store the return code and avoid this redundant call.

Fixing it now.